### PR TITLE
Multi arch

### PR DESCRIPTION
**Pull request by finkandreas**
This PR adds supports for multi-arch support.
We can build the containers on GH nodes, and on x86_64 nodes. They first get tagged with the appropriate architecture.
Then there is an additional script, that combines the two architecture specific images into a single manifest.
Effectively this would work on both an ARM machine and a x86_64 machine:
```
podman pull finkandreas/spack:0.20.3-ubuntu22.04
````

The arch specific images could be pulled via
```
podman pull finkandreas/spack:0.20.3-ubuntu22.04-x86_64
podman pull finkandreas/spack:0.20.3-ubuntu22.04-aarch64
```

This change will make sure that we are backwards compatible, i.e. old pipelines will continue to run and pick up the x86_64 architecture, while pipelines on GH will work too with the same image name as base container starting point.

We cannot have Rocm images because there are no base images for arm64.

### CODE REVIEW

**Comment by finkandreas** [comment]
One thing I do not like is that we have in the build script and the manifest creation script the same loop variables and must be maintained together. I see in the future that this might create some bugs on our side.
I am open for suggestions how this could be worked around. Separate file with the possible variable values and source the file?

**Comment by rjanalik** [comment]
Yeah, I also don't like this. I think we should move the variables to a file as you suggest, e.g., config.json.

File: build_and_push.sh [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+
+for spackver in "v0.20.3" "v0.21.0" ; do
+    for baseimg in docker.io/ubuntu:22.04 ; do
+        SPACK_DOCKER_TAG=$(echo $spackver | sed -e 's/^v//')
+        OS_DOCKER_TAG=$(basename "$baseimg" | sed -e 's/://')
+        DOCKER_TAG=${REMOTE}/spack:${SPACK_DOCKER_TAG}-${OS_DOCKER_TAG}
+        BASE_TAG_NAME=${REMOTE}/spack:base-${OS_DOCKER_TAG}
+        if [[ "$1" == "image" ]] ; then
```
Comment (rjanalik) [inline build_and_push.sh]: What about something like this?
```suggestion
        if [[ "$COMMAND" == "image" ]] ; then
```
And somewhere above
```
COMMAND=$1
```

File: build_and_push.sh [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            build_image "docker/Dockerfile_spack_baseimage_${OS_DOCKER_TAG}" "${DOCKER_TAG}-$(uname -m)" "--build-arg BASEIMG=$baseimg" "--build-arg SPACK_VER=$spackver"
+            build_image "docker/Dockerfile_base_helper" "${BASE_TAG_NAME}-$(uname -m)" "--build-arg BASEIMG=$baseimg"
+        elif [[ "$1" == "manifest" ]] ; then
```
Comment (rjanalik) [inline build_and_push.sh]: The same here
And 3x below

**Review by rjanalik — CHANGES_REQUESTED** [review]

**Review by rjanalik — APPROVED** [review]

**Pull request merged**